langwatch/prisma/migrations/20260728120000_index_apikey_for_session_key_reap/migration.sql

```
@@ -0,0 +1,40 @@
+-- 20260426150000_drop_redundant_audit_log_org_index, which hit the same wall).
+-- A plain CREATE INDEX takes a SHARE lock on ApiKey: reads keep running, WRITES
+-- BLOCK for the whole build. That is not hypothetical traffic — `ApiKeyService
+-- .markUsed` fires a fire-and-forget `apiKey.update({ lastUsedAt })` after every
+-- successful API-key authorization, including trace ingest (collector.ts,
+-- otel.ts), and `apiKey.create` runs on every Langy session-key mint. All of
+-- them stall until the index is built. On a large deployment, build it by hand
+-- with CREATE INDEX CONCURRENTLY first (same name, same definition) and this
+-- becomes a no-op via IF NOT EXISTS.
+CREATE INDEX IF NOT EXISTS "ApiKey_name_expiresAt_unrevoked_idx"
+  ON "ApiKey"("name", "expiresAt")
+  WHERE "revokedAt" IS NULL;
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)